datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -2402,4 +2454,757 @@ mod tests {
+
+        // Sort with batch_size = 100
+        let result_batches = sort_batch_chunked(&batch, &expressions, 100)?;
+
+        // Should return exactly 10 batches of 100 rows each
+        assert_eq!(result_batches.len(), 10);
+        for batch in &result_batches {
+            assert_eq!(batch.num_rows(), 100);
+        }
+
+        // Verify sorted correctly across all batches
+        let concatenated = concat_batches(&schema, &result_batches)?;
+        let array = as_primitive_array::<Int32Type>(concatenated.column(0))?;
+        for i in 0..array.len() - 1 {
+            assert!(array.value(i) <= array.value(i + 1));
+        }
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_sort_batch_chunked_with_nulls() -> Result<()> {
+        let schema = Arc::new(Schema::new(vec![Field::new("a", DataType::Int32, true)]));
+
+        // Create a batch with nulls
+        let values = Int32Array::from(vec![
+            Some(5),
+            None,
+            Some(2),
+            Some(8),
+            None,
+            Some(1),
+            Some(9),
+            None,
+            Some(3),
+            Some(7),
+        ]);
+        let batch = RecordBatch::try_new(Arc::clone(&schema), vec![Arc::new(values)])?;
+
+        // Test with nulls_first = true
+        {
+            let expressions: LexOrdering = [PhysicalSortExpr {
+                expr: Arc::new(Column::new("a", 0)),
+                options: SortOptions {
+                    descending: false,
+                    nulls_first: true,
+                },
+            }]
+            .into();
+
+            let result_batches = sort_batch_chunked(&batch, &expressions, 4)?;
+            let concatenated = concat_batches(&schema, &result_batches)?;
+            let array = as_primitive_array::<Int32Type>(concatenated.column(0))?;
+
+            // First 3 should be null
+            assert!(array.is_null(0));
+            assert!(array.is_null(1));
+            assert!(array.is_null(2));
+            // Then sorted values
+            assert_eq!(array.value(3), 1);
+            assert_eq!(array.value(4), 2);
+        }
+
+        // Test with nulls_first = false
+        {
+            let expressions: LexOrdering = [PhysicalSortExpr {
+                expr: Arc::new(Column::new("a", 0)),
+                options: SortOptions {
+                    descending: false,
+                    nulls_first: false,
+                },
+            }]
+            .into();
+
+            let result_batches = sort_batch_chunked(&batch, &expressions, 4)?;
+            let concatenated = concat_batches(&schema, &result_batches)?;
+            let array = as_primitive_array::<Int32Type>(concatenated.column(0))?;
+
+            // First should be 1
+            assert_eq!(array.value(0), 1);
+            // Last 3 should be null
+            assert!(array.is_null(7));
+            assert!(array.is_null(8));
+            assert!(array.is_null(9));
+        }
+
+        Ok(())
+    }
```

Comment:
Is this test relevant? it seems like it tests the sort itself rather then the chunked